aiida/cmdline/utils/migration/v01_to_v02.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida_core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Migration from v0.1 to v0.2, used by `verdi export migrate` command."""
+# pylint: disable=unused-argument
+from __future__ import division
+from __future__ import print_function
+from __future__ import absolute_import
+
+from aiida.cmdline.utils.migration.utils import verify_metadata_version, update_metadata
+
+__all__ = ('migrate_v1_to_v2',)
+
+
+def migrate_v1_to_v2(metadata, data, *args):
+    """
+    Migration of export files from v0.1 to v0.2, which means generalizing the
+    field names with respect to the database backend
+
+    :param metadata: the content of an export archive metadata.json file
+    :param data: the content of an export archive data.json file
+    """
+    old_version = '0.1'
+    new_version = '0.2'
+
+    old_start = "aiida.djsite"
+    new_start = "aiida.backends.djsite"
+
+    verify_metadata_version(metadata, old_version)
+    update_metadata(metadata, new_version)
+
+    def get_new_string(old_string):
+        """Replace the old module prefix with the new."""
+        if old_string.startswith(old_start):
+            return '{}{}'.format(new_start, old_string[len(old_start):])
+
+        return old_string
+
+    def replace_requires(data):
+        """Replace the requires keys with new module path."""
+        if isinstance(data, dict):
+            new_data = {}
+            for key, value in data.items():
+                if key == 'requires' and value.startswith(old_start):
+                    new_data[key] = get_new_string(value)
+                else:
+                    new_data[key] = replace_requires(value)
+            return new_data
+
+        return data
+
+    for field in ['export_data']:
+        for key in list(data[field]):
```

Comment:
what about replacing all this code with something like
```
data[field] = {  get_new_string(k) : v for k,v in data[field].items() }
```

this applies in other places, too

Reply:
This is the original function written by @sphuber. We have not changed it, only moved it to this new file.

I can try and change it?

Reply:
I see - unfortunately it's not straightforward for me to see what you have changed here and what has been moved.

It might still be worth streamlining some of this code where it is obvious and doesn't take much time.
Anyhow, up to you since it is not your code.

Reply:
> I see - unfortunately it's not straightforward for me to see what you have changed here and what has been moved.

No. The only way would be to not look at all the commits at the same time, but go back to before the commit, where I moved everything around (which I have left as a single commit, and should probably be left like this in the final merge as well).

> It might still be worth streamlining some of this code where it is obvious and doesn't take much time.
> Anyhow, up to you since it is not your code.

I will take a look at it, but now have it at the top of my priority queue :)

Reply:
I tried to change this, also for `migrate_v2_to_v3`, but it seemed my tests became slower. Maybe it is the memory usage required by the single-line commands? In any case, I will not touch these functions at this point.

Reply:
Ok, fine